### ISSUE TITLE
Stats Servlet hidden from classpath

### ISSUE THREAD

**Issue by gregw** [eclipse/jetty.project#1128]

**Comment by gregw** [comment]
I have pushed a work around.  However, we should review if the StatsServlet should be in the o.e.j.servlets package instead?

**Comment by sbordet** [comment]
@gregw the `jetty-servlets` module is being used as a dependency by web applications that want to use its portable filters/servlets, typical examples being `CrossOriginFilter`, `DoSFilter`, `EventSourceServlet`, etc. These "utility" filters/servlets may depend on `jetty-util` or `jetty-http` but that's fine.

On the other hand, `PushCacheFilter`, `StatisticsServlet`, etc. use server classes such as `Request`, `Connector`, etc. so they are not portable.

To be precise we should have put `PushCacheFilter` into `jetty-servlet`.
`PushCacheFilter` is a special case in that it uses a proposed API that should become standard (and only once the proposed API became a standard, we should have moved it to `jetty-servlets`).

Since `StatisticsServlet` uses so many Jetty server classes and it's not portable, I think it should remain in `jetty-servlet` and we should punch a hole for it, like your workaround does.

**Comment by dfish3r** [comment]
@gregw did you confirm this works?
Adding
```
  <servlet>
    <servlet-name>Status</servlet-name>
    <servlet-class>org.eclipse.jetty.servlet.StatisticsServlet</servlet-class>
    <init-param>
      <param-name>restrictToLocalhost</param-name>
      <param-value>false</param-value>
    </init-param>
    <load-on-startup>1</load-on-startup>
  </servlet>
```

Produces: `java.lang.ClassNotFoundException: org.eclipse.jetty.servlet.StatisticsServlet` on version 9.4.2.v20170220 for my simple webapp.

Since all the other uses of `jetty.webapp.addServerClasses` add library paths and not classnames, I wonder if the usage is correct.

**Comment by joakime** [comment]
@dfish3r do you have `--module=servlets` enabled in your `${jetty.base}`?

**Comment by dfish3r** [comment]
My `start.ini` in `${jetty.base}` includes
```
--module=stats
--module=servlets
```

I also tried adding `--module=servlet`, which I didn't think was necessary.
Nothing has worked thus far.

**Comment by janbartel** [comment]
@dfish3r

**Comment by dfish3r** [comment]
@janbartel any update on this?
If the `StatisticsServlet` isn't meant for consumption by other webapps, that's fine.
Just noting that the behavior changed somewhere along the line from 9.3 to 9.4

**Comment by janbartel** [comment]
@dfish3r I haven't been able to reproduce the problem.

I'm using jetty 9.4.x head.  I used the standard jetty test webapp. All I did was:

 Copy your servlet definition into web.xml

```java -jar ../start.jar --add-to-start=stats
```

Then run and it worked fine.

Can you provide more details on exactly what you are doing?

**Comment by janbartel** [comment]
@dfish3r  and just for completeness, I also tested 9.4.2.v20170220 and got the same result, ie it works. Need more info.

**Comment by gregw** [comment]
See #1384.   We had inadvertently hidden the servlet.  Fixed in next releases of 9.3. and 9.4 out in the next few days.

**Comment by janbartel** [comment]
@gregw no, the stats.mod file already explicitly exposed the StatisticsServlet as a server class, so you need to undo whatever you just did.

@dfish3r I think your problem is that you just added --module=stats to your start.ini without actually running the -jar --add-to-start command:  that won't work because there is module ini template that needs to be generated (ie installing the module does more than just adding a --module line to start.ini).

**Comment by gregw** [comment]
Fixed and/or explained again in #1384

**Comment by dfish3r** [comment]
Thanks @janbartel. You are correct. I needed to add `jetty.webapp.addServerClasses+=,-org.eclipse.jetty.servlet.StatisticsServlet` to my start.ini
